### PR TITLE
Accomodate Seq in bind_expr, fission_after, add_loop, double fission

### PR DESCRIPTION
**Pull request by yamaguchi1024**

### CODE REVIEW

File: src/exo/LoopIR_scheduling.py [hunk 1]
```diff
@@ -1092,7 +1092,7 @@ def map_s(self, s):
 
         if isinstance(s, (LoopIR.ForAll, LoopIR.Seq)):
             body = self.process_block(s.body)
-            return [LoopIR.ForAll(s.iter, s.hi, body, s.eff, s.srcinfo)]
+            return [type(s)(s.iter, s.hi, body, s.eff, s.srcinfo)]
```
Comment (alexreinking) [inline src/exo/LoopIR_scheduling.py:1095]: Use `s.update(body=body)`

**Review by alexreinking — CHANGES_REQUESTED** [review]

**Comment by alexreinking** [comment]
We have a helper for this. Might as well use it.

File: src/exo/LoopIR_scheduling.py [hunk 2]
```diff
@@ -2154,17 +2154,17 @@ def map_s(self, s):
                 # body doesn't depend on the loop
                 # and the body is idempotent
                 if s.iter in _FV(pre) or not _is_idempotent(pre):
-                    pre     = [LoopIR.ForAll(s.iter, s.hi, pre, None, s.srcinfo)]
+                    pre     = [type(s)(s.iter, s.hi, pre, None, s.srcinfo)]
```
Comment (alexreinking) [inline src/exo/LoopIR_scheduling.py:2157]: `s.update(body=pre, eff=None)`

**Comment by yamaguchi1024** [comment]
Hmm isn't this okay? The helper is not used in other parts of this file

**Comment by alexreinking** [comment]
> Hmm isn't this okay? The helper is not used in other parts of this file

I didn't retroactively replace all of these with the helper, but that doesn't mean we shouldn't use it in new code.

**Comment by yamaguchi1024** [comment]
I'll fix this ASAP!

**Review by alexreinking — APPROVED** [review]

**Pull request merged**